### PR TITLE
effects: simplify linked lists management

### PR DESCRIPTION
**Pull request by rr-**
The next_fx was split into a draw and a free pointer, which makes it easier to understand.

#### Checklist

- [X] I have read the [coding conventions](https://github.com/rr-/Tomb1Main/blob/master/CONTRIBUTING.md#coding-conventions)
- [X] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description

The next_fx was split into a draw and a free pointer, which makes it easier to understand. Originally I had thought this was an OG bug but actually it seems to be working fine, so it's just a cleanup PR.

### CODE REVIEW

**Comment by walkawayy** [comment]
Good changes but I'll wait to test until flame emitter fix is merged.

**Comment by rr-** [comment]
Thanks. I've merged the other branches and rebased this onto the newest develop.

**Review by walkawayy — APPROVED** [review]
Worked well for me. I tested saving and loading combinations in Lahm's test level with lava and flames in all the different rooms, some underwater bubbles, waterfall mist, and Midas flame emitters.

**Comment by rr-** [comment]
Waiting for @lahm86 before I merge

**Review by lahm86 — APPROVED** [review]

**Pull request merged**
